### PR TITLE
Check for NIOS cloud license

### PR DESCRIPTION
**Pull request by AjeyHiremath**

### CODE REVIEW

File: daemon/license.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+func CheckLicense(objMgr *ibclient.ObjectManager, licenseType string) (flag bool, err error) {
+	license, err := objMgr.GetLicense()
+	if err != nil {
+		return flag, err
+	}
+	for _, v := range license {
+		if strings.ToLower(v.Licensetype) == licenseType {
+			if v.ExpirationStatus != "DELETED" && v.ExpirationStatus != "EXPIRED" {
+				flag = true
```
Comment (chinmayb) [inline daemon/license.go]: is the flag necessary ? Cant we change it to whatever we discussed on friday ?

Comment (AjeyHiremath) [reply daemon/license.go]: When testing that was not working, As we discussed, We were checking conditions only if the cloud type is cloud and if there is no cloud type is cloud then we were doing nothing so it was not working as expected

File: daemon/license.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+			}
+		}
+	}
+	return flag, err
```
Comment (chinmayb) [inline daemon/license.go]: err = fmt.Errorf("%s License not available/applied. Apply the license and try again", licenseType)
return err

File: daemon/license.go (hunk 1 above)
Comment (chinmayb) [reply daemon/license.go]: still flag isnt necessary.  can you please try the other way i asked ?

Comment (AjeyHiremath) [reply daemon/license.go]: done

File: daemon/license.go (hunk 2 above)
Comment (AjeyHiremath) [reply daemon/license.go]: done

File: daemon/license.go [hunk 3]
```diff
@@ -9,26 +10,24 @@ import (
...
 //Checks for cloud license in nios
 func CheckForCloudLicense(objMgr *ibclient.ObjectManager) {
-	flag, err := CheckLicense(objMgr, "cloud")
+	err := CheckLicense(objMgr, "cloud")
 	if err != nil {
-		logrus.Fatal("error", err)
-	}
-	if !flag {
-		logrus.Fatal("Cloud License not available in Infoblox Appliance. Update and try again..")
+		logrus.Fatal("Check Cloud License : ", err)
```
Comment (chinmayb) [inline daemon/license.go]: replace to "Error while checking for cloud license: "

File: daemon/license.go [hunk 4]
```diff
@@ -9,26 +10,24 @@ import (
...
 	for _, v := range license {
 		if strings.ToLower(v.Licensetype) == licenseType {
 			if v.ExpirationStatus != "DELETED" && v.ExpirationStatus != "EXPIRED" {
-				flag = true
+				return
 			}
 		}
 	}
-	return flag, err
+	err = fmt.Errorf("%s License not available/applied. Apply the license and try again", licenseType)
```
Comment (chinmayb) [inline daemon/license.go]: good to update to "Apply the license for the grid and try again"

File: daemon/license.go (hunk 3 above)
Comment (AjeyHiremath) [reply daemon/license.go]: done

File: daemon/license.go (hunk 4 above)
Comment (AjeyHiremath) [reply daemon/license.go]: done

**Pull request merged**